x-pack/plugins/infra/server/lib/alerting/metric_threshold/metric_threshold_executor.ts

```diff
@@ -246,6 +246,22 @@ export const createMetricThresholdExecutor = (libs: InfraBackendLibs) =>
         });
       }
     }
+
```

Comment:
@simianhacker Above you should schedule actions only for the active alerts, right?

```suggestion
const actionGroupId =
            nextState === AlertStates.NO_DATA
            ? NO_DATA_ACTIONS.id
            : nextState === AlertStates.WARNING
            ? WARNING_ACTIONS.id
            : FIRED_ACTIONS.id;
```

Reply:
Correct

Reply:
Do you still need to include 
```
nextState === AlertStates.OK
            ? RecoveredActionGroup.id
```

In the suggestion above, I suggested that you remove above check (similar to how you do it in Inventory PR)

Reply:
No because none of the items are `OK` in the `group` array. That array only contains items that are either `NO_DATA`, `WARNING`, or `FIRED`.